scb-scanprocesses/combined-nmap-sslyze-process/src/test/java/io/securecodebox/scanprocess/NmapToSslyzeTransformListenerTest.java

```diff
@@ -0,0 +1,70 @@
+package io.securecodebox.scanprocess;
+
+import io.securecodebox.model.execution.Target;
+import io.securecodebox.model.findings.Finding;
+import io.securecodebox.model.findings.OsiLayer;
+import io.securecodebox.model.findings.Reference;
+import io.securecodebox.model.findings.Severity;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class NmapToSslyzeTransformListenerTest {
+
+    private static NmapToSslyzeTransformListener listener;
+    private static List<Finding> findings;
+    private static final String TARGET_HOST = "test-host";
+    private static final String TARGET_LOCATION = "test-location";
+
+    @BeforeAll
+    private static void setUp() {
+        listener = new NmapToSslyzeTransformListener();
+        Finding finding1 = createFindingWithService("https");
+        Finding finding2 = createFindingWithService("ssl");
+        Finding finding3 = createFindingWithService("tls");
+        Finding finding4 = createFindingWithService("ssh");
+        findings = new ArrayList<Finding>();
+        findings.add(finding1);
+        findings.add(finding2);
+        findings.add(finding3);
+        findings.add(finding4);
+    }
+
+    @Test
+    protected void filteredServicesShouldBecomeNewTargets() {
+        assertEquals(3, listener.createTargetsFromFindings(findings).size(), "services are detected correctly");
```

Comment:
Would be cool to also have assertion to test that the target create have the correct locations

Reply:
added more specific tests